### PR TITLE
8353 Coop IA people and roles validation updates

### PR DESCRIPTION
**Pull request by argush3**
*Issue #:* /bcgov/entity#8353

*Description of changes:*

* Add validation rules for directors in an incorporation application filing
* Add/update tests to support new validation rules

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/services/filings/validations/incorporation_application.py [hunk 1]
```diff
@@ -94,17 +94,18 @@ def validate_offices(incorporation_json) -> Error:
     return None
 
 
+# pylint: disable=too-many-branches
 def validate_roles(incorporation_json) -> Error:
     """Validate the required completing party of the incorporation filing."""
     parties_array = incorporation_json['filing']['incorporationApplication']['parties']
+    legal_type = incorporation_json['filing']['incorporationApplication']['nameRequest']['legalType']
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/services/filings/validations/incorporation_application.py:101]: If you are trying to get legalType of business get it from '/filing/business/legalType' and I would suggest to use get_str function

File: legal-api/src/legal_api/services/filings/validations/incorporation_application.py [hunk 2]
```diff
@@ -139,8 +149,12 @@ def validate_roles(incorporation_json) -> Error:
 
 def validate_parties_mailing_address(incorporation_json) -> Error:
     """Validate the person data of the incorporation filing."""
+    legal_type = incorporation_json['filing']['incorporationApplication']['nameRequest']['legalType']
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/services/filings/validations/incorporation_application.py:152]: If you are trying to get legalType of business get it from '/filing/business/legalType' and I would suggest to use get_str function

**Review by vysakh-menon-aot — CHANGES_REQUESTED** [review]

File: legal-api/src/legal_api/services/filings/validations/incorporation_application.py (hunk 1 above)
Comment (argush3) [reply legal-api/src/legal_api/services/filings/validations/incorporation_application.py]: updated to grab legal type from `/filing/business/legalType` and use `get_str` function

File: legal-api/src/legal_api/services/filings/validations/incorporation_application.py (hunk 2 above)
Comment (argush3) [reply legal-api/src/legal_api/services/filings/validations/incorporation_application.py]: updated to grab legal type from `/filing/business/legalType` and use `get_str` function

**Review by andrepestana-aot — APPROVED** [review]
LGTM

**Review by vysakh-menon-aot — APPROVED** [review]

**Pull request merged**